### ISSUE TITLE
update --warn      (instant server shutdown)

### ISSUE THREAD

**Issue by kyle142** [FezVrasta/ark-server-tools#442]
When running 

*/15 * * * * /usr/local/bin/arkmanager update --update-mods --warn --verbose

The server instantly shuts down when the update has been triggered without warning. 

Is anyone else experiencing this?

**Comment by klightspeed** [comment]
Assuming you haven't set up cron to email you the output of cron jobs, is there anything in /home/steam/dead.letter or /var/spool/mail/steam?  (emails go to `/var/spool/mail/<username>` by default on most systems, and `~/dead.letter` on systems without a mailserver set up).

Does the number of players the `Players:` line says are connected match how many are actually connected?

**Comment by kyle142** [comment]
Thanks for your reply :) I checked `/var/spool/mail/nobody` and saw no mention of arkmanager.

I do have a feeling that i remember seeing something like inaccurate reporting of players.. from reporting different results of `arkmanager status` and `arkmanager rconcmd "ListPlayers"`

**Comment by klightspeed** [comment]
So was `nobody` the only file in `/var/spool/mail`?

Perhaps try redirecting the output of the script to a log file?

The reason I asked about the `Players:` count is that the `--warn` implementation in arkmanager will immediately proceed with the operation if the player count equals `0`.

#444 should fix the case of immediately shutting down if the server doesn't respond quickly on its QueryPort.

**Comment by arkfrancegaming** [comment]
So is it possible to add in the cron list the command arkmanager update --update-mods --warn
to check every x minutes if there are some mod updates ?
but if there are no updates, does it always restart the server ?

**Comment by kyle142** [comment]
@arkfrancegaming
Thats right, works exactly as you said, only updates\restarts if theres an update - It's just I'm having this issue is all :)

**Comment by arkfrancegaming** [comment]
does the warn works fine now ?

**Comment by kyle142** [comment]
I believe this issue is only affecting me. Waiting for the next update to
generate a log

**Comment by arkfrancegaming** [comment]
i'm gonna try to check this out

**Comment by klightspeed** [comment]
@kyle142 you should be able to fake there being an update available by either removing `$arkserverroot/steamapps/appmanifest_376030.acf` or modifying it to change the buildid.

**Comment by kyle142** [comment]
Hey, you got it.
`This ARK server will shutdown for an update in 30 minutes
Server received, But no response!!

Nobody is connected.  Shutting down immediately
Stopping server...
The server has been stopped
`

That was while 2 players were connected. Rcon is enabled, with port. `arkmanager status` and `arkmanager rconcmd "ListPlayers"` reports the correct number. with 2 people currently connected

**Comment by arkfrancegaming** [comment]
i don't know about you but sometimes the serveur restart for a mod update while there is no new updates available

**Comment by kyle142** [comment]
@arkfrancegaming. Try the same trouble shooting that Ben recommended myself to see if you have the same problem, if you don't you should make a new thread so we can help :)

May not be the problem, but you're checking for mod updates every 2 minutes, depending on how many mods you have and other factors, you may be in a constant update loot?  Try 10, 15, whatever.  It may not be the problem but I'm thinking the first update doesn't complete before the 2 minutes is up

**Comment by arkfrancegaming** [comment]
i'm doing a check every 15 min

**Comment by kyle142** [comment]
Just be  sure you're running  1.5v of arkmanager tools

**Comment by arkfrancegaming** [comment]
i'm running 1.6, but i've got 7 servers, it only happen with just one

**Comment by kyle142** [comment]
Running v1.6 of ark server tools I still get the same message ```
Nobody is connected.  Shutting down immediately
Stopping server...
The server has been stopped

```

**Comment by klightspeed** [comment]
@kyle142 found the issue - `numPlayersConnected` was still using the `ark_QueryPort` setting where it should have been using the `getQueryPort` function.

I have created a `1.6-dev#unstable` channel with the fixes for this in it.  Please let me know if they fix the issue.

**Comment by arkfrancegaming** [comment]
is this can be related to my pb too ? @klightspeed

**Comment by klightspeed** [comment]
@arkfrancegaming I will work on unpacking mods to a staging directory and then comparing them to the installed mods.  I suspect steamcmd might be changing the modification time on files in the mod workshop directory even though their contents haven't changed.

**Comment by arkfrancegaming** [comment]
Might be good, the weird thing is it happens randomly only for one mod in one server

**Comment by klightspeed** [comment]
@arkfrancegaming Is this mod used by any of the other instances on the server?

**Comment by arkfrancegaming** [comment]
This mod is Valhalla, nope it's only on one instance

**Comment by kyle142** [comment]
Trying to move to branch 1.6-dev#unstable
```
A new version was found! Do you want to upgrade ARK Server Tools to v400: Invalid request?y
bash: line 1: 400:: command not found
```

**Comment by klightspeed** [comment]
@kyle142 try `1.6-dev.unstable`

**Comment by arkfrancegaming** [comment]
is it possible to just check the update not download it of there is any ?

**Comment by kyle142** [comment]
You can view the list of commands in the readme, or type arkmanager help in
the terminal to give a list of commands. Checkupdate

**Comment by arkfrancegaming** [comment]
i know but i mean the command it self, can it just check instead of downloading everytime ?

**Comment by kyle142** [comment]
Arkmanager checkupdate

**Comment by arkfrancegaming** [comment]
i mean for the mod update

**Comment by klightspeed** [comment]
The only way to check for mod updates is to actually download them.  Try `arkmanager update @instance --update-mods --downloadonly` - this downloads the mod updates but does not apply them.

**Comment by arkfrancegaming** [comment]
i've search for my pb, it seems it find an update only for the .modbranch and the .mod in the mod folder

**Comment by klightspeed** [comment]
@arkfrancegaming those two files are constructed by the mod update procedure.  The `.mod` file is not used to determine if an update is needed, and the `.modbranch` file only triggers an update if its contents don't match the `mod_branch` variable.

Check in `$steamcmdroot/steamapps/workshop/content/346110/504122600/WindowsNoEditor`

**Comment by arkfrancegaming** [comment]
the weird part is that i've another server with the same mod and it doesn't do that, i've copy those two file from the good server to this one to check if it changes something

**Comment by arkfrancegaming** [comment]
those two file were missing in steamcmdroot/steamapps/workshop/content/346110/504122600/

**Comment by arkfrancegaming** [comment]
Still does the same :(

**Comment by kyle142** [comment]
without logs it's really hard to say

**Comment by arkfrancegaming** [comment]
update mod logs ?

**Comment by klightspeed** [comment]
`~/Steam/logs/workshop_log.txt`

**Comment by arkfrancegaming** [comment]
i don't think so because i've got no workshop_log.txt in this folder

**Comment by klightspeed** [comment]
@arkfrancegaming The `/home/$steamcmduser/Steam/logs/workshop_log.txt` file is created by steamcmd when updating mods.  

Are there any files under `/home/<user>/Steam/logs` where `<user>` is the user under which the ARK server runs?

**Comment by arkfrancegaming** [comment]
[2016-03-12 12:27:35] [AppID 346110] Loaded workshop items in "/home/steam/steamapps/workshop" (15 installed, 15 needed)
[2016-03-12 12:27:35] [AppID 346110] Starting Workshop download job
[2016-03-12 12:27:37] Update workshop job for app 346110 canceled: Failed to move file /home/steam/steamapps/workshop/content/346110/504122600/LinuxNoEditor/Custom_Assets/Dungeon/Textures/Sewer/Sewerage_D.uasset.z.uncompressed_size (Disk read failure)
[2016-03-12 12:27:37] [AppID 346110] Download item 599476919 result : I/O Operation Failed
[2016-03-12 12:27:37] [AppID 346110] Finished Workshop download job : Disk read failure
[2016-03-12 12:27:37] [AppID 346110] Updating workshop item details failed with Failure

**Comment by arkfrancegaming** [comment]
might be that

**Comment by klightspeed** [comment]
Do you have any kernel errors around that time?

Also, check the permissions on `/home/steam/steamapps/workshop/content/346110/504122600/LinuxNoEditor/Custom_Assets/Dungeon/Textures/Sewer/Sewerage_D.uasset.z.uncompressed_size` and `/home/steam/steamapps/workshop/downloads/346110/504122600/LinuxNoEditor/Custom_Assets/Dungeon/Textures/Sewer/Sewerage_D.uasset.z.uncompressed_size`

**Comment by arkfrancegaming** [comment]
Permissions are 775 for all

**Comment by arkfrancegaming** [comment]
i'm going to remove all the 504122600 folder and re download it to check

**Comment by kyle142** [comment]
Hey @arkfrancegaming infutre can you please not hijack the thread? I know youre asking for help buddy but my phones going off during the day in my pocket for your issue thats unrelated to mine and it also makes the TWO issues hard to follow when reading here, best in-future to make 1 thread per issue.


@klightspeed changing the branch from `"master"` to `"1.6-dev"`
and running `arkmanager upgrade-tools` update doesn't go smoothly, is there somewhere I can few the logs as to why the tool upgrade is failing so i can fix this before I'm able to test`1.6-dev.unstable`? Thanks for all your help

**Comment by kyle142** [comment]
1.6.dev.unstable   <-- does NOT solve the problem

**Comment by klightspeed** [comment]
@kyle142 what is the output of the script when it fails to go through the whole warning period?

**Comment by kyle142** [comment]
Apologies, I just realized that "arkmanager" was using version 1.5, however
"sudo arkmanager" was using 1.6, after moving my cron tab over to root - I
can confirm 1.6.dev.unstable fixes the issue of instant disconnect during
update without warning :-). Thank you.

Also meaning to send you some donation money if possible? But I can't see
how to private message you
